server/xpub-client/pubsub/index.js

```diff
@@ -0,0 +1,29 @@
+class PubsubClient {
+  constructor(PubsubManager) {
+    this.manager = PubsubManager
+    this.pubsub = null
+    this.publishingInterval = null
+  }
+
+  async initialize() {
+    this.pubsub = await this.manager.getPubsub()
+  }
+
+  startPublishingOnInterval(message, dataCallback, interval) {
+    this.publishingInterval = setInterval(
+      () => this.pubsub.publish(message, dataCallback()),
+      interval,
+    )
+  }
+
+  stopPublishingOnInterval(message, dataCallback) {
+    clearInterval(this.publishingInterval)
+    this.publishingInterval = null
+
+    if (message && dataCallback) {
+      this.pubsub.publish(message, dataCallback())
```

Comment:
Client consumers may want to call publish on its own without an interval. Could we create a `publish` function on the object that just calls `this.pubsub.publish` so that calling it is less verbose